getssl

```diff
@@ -641,6 +642,11 @@ create_csr() { # create a csr using a given key (if it doesn't already exist)
     if [[ "$OCSP_MUST_STAPLE" == "true" ]]; then
       printf "\n1.3.6.1.5.5.7.1.24 = DER:30:03:02:01:05" >> "$tmp_conf"
     fi
+    # this is required if you want to use LDAP over SSL in Microsoft Active Directory
+    # see also: https://serverfault.com/a/573038 and https://support.microsoft.com/en-us/help/321051/
+    if [[ "$MS_LDAPS" == "true" ]]; then
+      printf "\nextendedKeyUsage = 1.3.6.1.5.5.7.3.1" >> "$tmp_conf"
+    fi
```

Comment:
:wave: hey @tohn,

I'm not a developer of `getssl` but I saw this PR and I'm a little bit confused.

The EKU OID you're including here isn't specific to LDAP and MS AD. It's the bog-standard "TLS WWW Server Authentication" OID, referred to as `"id-kp-serverAuth"` in [RFC 5280](https://tools.ietf.org/html/rfc5280). I think you'll be hard-pressed to find a CA that issues HTTPS certificates that don't include this OID. As one concrete example Let's Encrypt includes it in every certificate we issue.

You can double check that's the case by ref'ing RFC 5280. The oid `1.3.6.1.5.5.7.3.1` is:
`id-pkixk` (1.3.6.1.5.5.7) + `id-kp` (3) + `id-kp-serverAuth` (1) = 1.3.6.1.5.5.7.3.1

Speaking about Let's Encrypt in particular we totally ignore the EKU's in the CSR and always assign a static set based on the type of public key the certificate describes. e.g. if there is some other special OID you need for LDAP/MS AD you **won't** be able to get Let's Encrypt to include it in a certificate by putting it into the CSR. Other CAs might, I can't speak to them.